src/fides/api/ops/models/privacy_notice.py

```diff
@@ -0,0 +1,216 @@
+from __future__ import annotations
+
+from collections import defaultdict
+from enum import Enum
+from typing import Any, Dict, Iterable, List, Tuple, Type
+
+from sqlalchemy import Boolean, Column
+from sqlalchemy import Enum as EnumColumn
+from sqlalchemy import Float, ForeignKey, String
+from sqlalchemy.dialects.postgresql import ARRAY
+from sqlalchemy.orm import Session, relationship
+
+from fides.api.ops.common_exceptions import ValidationError
+from fides.lib.db.base_class import Base, FidesBase
+
+
+class PrivacyNoticeRegion(Enum):
+    """
+    Enum is not formalized in the DB because it is subject to frequent change
+    """
+
+    us_ca = "us_ca"
+    us_co = "us_co"
+    us_va = "us_va"
+    us_ut = "us_ut"
+    eu = "eu"
+
+
+class ConsentMechanism(Enum):
+    opt_in = "opt_in"
+    opt_out = "opt_out"
+    necessary = "necessary"
+
+
+class EnforcementLevel(Enum):
+    """
+    Enum is not formalized in the DB because it may be subject to frequent change
+    """
+
+    frontend = "frontend"
+    system_wide = "system_wide"
+
+
+class PrivacyNoticeBase:
+    """
+    Base class to establish the common columns for `PrivacyNotice`s and `PrivacyNoticeHistory`s
+    """
+
+    name = Column(String, nullable=False)
+    description = Column(String, nullable=True)
+    origin = Column(String, nullable=True)  # pointer back to an origin template ID
+    regions = Column(
+        ARRAY(EnumColumn(PrivacyNoticeRegion, native_enum=False)),
+        index=True,
+        nullable=False,
+    )
+    consent_mechanism = Column(EnumColumn(ConsentMechanism), nullable=False)
+    data_uses = Column(
+        ARRAY(String), nullable=False
+    )  # a list of `fides_key`s of `DataUse` records
+    enforcement_level = Column(EnumColumn(EnforcementLevel), nullable=False)
+    version = Column(Float, nullable=False, default=1.0)
+    disabled = Column(Boolean, nullable=False, default=False)
+    has_gpc_flag = Column(Boolean, nullable=False, default=False)
+    displayed_in_privacy_center = Column(Boolean, nullable=False, default=True)
+    displayed_in_banner = Column(Boolean, nullable=False, default=True)
+    displayed_in_privacy_modal = Column(Boolean, nullable=False, default=True)
+
+
+class PrivacyNotice(PrivacyNoticeBase, Base):
+    """
+    A notice set up by a system administrator that an end user (i.e., data subject)
+    accepts or rejects to indicate their consent for particular data uses
+    """
+
+    @classmethod
+    def create(
+        cls: Type[PrivacyNotice],
+        db: Session,
+        *,
+        data: dict[str, Any],
+        check_name: bool = True,
```

Comment:
Seems like the default here should be false?

Reply:
yes! thank you, good catch